app/analytics/query.ts

```diff
@@ -188,6 +203,7 @@ export class AnalyticsEngineAPI {
 
         // note interval count hard-coded to hours at the moment
         const initialRows = generateEmptyRowsOverInterval(
+            interval,
```

Comment:
It strikes me that you're passing in `interval` into a few functions to strictly do `if (interval) === "yesterday"`.

I think it would be better to just pass an optional `endDateTime`, just as `startDateTime` is being passed. Basically, determine the value in `dashboard.tsx` and pass it in. That way any arbitrary `endDateTime` could be used.

Thoughts?

Reply:
That's a good point. I'll refactor

Reply:
Much cleaner

Reply:
Only part I'm not 100% on is the placement of the parameter. I like it grouped with the startDateTime and that makes sense but it looks off when you want to leave it out. 

```ts
            const result1 = await api.getViewsGroupedByInterval(
                "example.com",
                "DAY",
                new Date("2024-01-11 00:00:00"), // local time (because tz also passed)
                undefined,
                "America/New_York",
            );
```

This is from a test case for example. Not sure how I feel about the undefined.